pytorch_lightning/trainer/trainer.py

```diff
@@ -1307,7 +1307,13 @@ def _run_sanity_check(self, ref_model):
         using_val_step = self._data_connector._val_dataloader_source.is_defined() and is_overridden(
             "validation_step", ref_model
         )
-        should_sanity_check = using_val_step and self.num_sanity_val_steps > 0 and self.limit_val_batches > 0
+        should_sanity_check = (
+            using_val_step
+            and self.num_sanity_val_steps > 0
+            and self.limit_val_batches > 0
+            # do not sanity check if restarting because it would mess up the loaded state
+            and not self._evaluation_loop.restarting
```

Comment:
mind add a test?

Reply:
This will be necessary for #8578 which includes many tests. I've split this single change as it's a decision that wasnt discussed before